examples/basecoin/cmd/basecoind/main.go

```diff
@@ -1,10 +1,13 @@
 package main
 
-import "github.com/cosmos/cosmos-sdk/examples/basecoin/app"
+import (
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/examples/basecoin/app"
+)
 
 func main() {
 	// TODO CREATE CLI
 
 	bapp := app.NewBasecoinApp("")
-	bapp.RunForever()
+	baseapp.RunForever(bapp)
```

Comment:
bapp.RunForever()?

Reply:
This is the simplest way to achieve a reusable RunForever function (found in `baseapp/helpers.go`). Don't really see another easy to inherit from baseapp because `server.NewServer("0.0.0.0:46658", "socket", app)` takes the actual abci.Application as an input